sqlmesh/core/model/kind.py

```diff
@@ -341,6 +348,24 @@ def managed_columns(self) -> t.Dict[str, exp.DataType]:
         }
 
 
+class SCDType2ByTimeKind(_SCDType2Kind):
+    name: Literal[ModelKindName.SCD_TYPE_2_BY_TIME] = ModelKindName.SCD_TYPE_2_BY_TIME
+    updated_at_name: SQLGlotString = "updated_at"
+    updated_at_as_valid_from: SQLGlotBool = False
+
+
+# Legacy alias to original SCD Type 2 Kind that existed prior to adding by column support
+# Only used in discriminator for Pydantic
+class SCDType2Kind(SCDType2ByTimeKind):
+    name: Literal[ModelKindName.SCD_TYPE_2] = ModelKindName.SCD_TYPE_2_BY_TIME  # type: ignore
```

Comment:
Why don't the literal type and the value match?

Reply:
Thanks for your feedback and helping me to dig deeper into this. I removed what we talked about (including this line) but then tested actually loading SCD_TYPE_2 from state. This commit fixes the issues identified: https://github.com/TobikoData/sqlmesh/pull/1997/commits/9a76e6fdf610492c6c004e13356548850ca1bbb3

Btw the reason why I left the default values at all is because it makes it easy in testing to do `FullKind()` and let the default put in full. We could always create a fixture that properly constructs models with the name included but it felt like I was going down the rabbit hole a bit and stopped here.